src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java

```diff
@@ -1488,7 +1490,8 @@ private void handlePolicyRemovalEvent(MonitorPolicyEvent event, Monitor clonedMo
     List<MonitorPolicyDTO> effectivePolicies = policyApi.getEffectiveMonitorPoliciesForTenant(tenantId, false);
 
     // if the old policy had been overriding another one using the same monitorId, find the other one
-    List<UUID> newPolicyIds = effectivePolicies.stream()
+    List<UUID>  newPolicyIds = effectivePolicies.stream()
+        .filter(p -> event.getMonitorId() != null)
         .filter(p -> p.getMonitorId().equals(monitorId))
```

Comment:
...or could use the null-safe approach and avoid the need for the added filter above
```suggestion
        .filter(p -> Objects.equals(p.getMonitorId(), monitorId))
```